sys/console/minimal/src/console.c

```diff
@@ -86,11 +86,17 @@ console_read(char *str, int cnt, int *newline)
     }
     cmd = ev->ev_arg;
     len = strlen(cmd->line);
+
     if ((cnt - 1) < len) {
         len = cnt - 1;
+        if (len > 0) {
+            memcpy(str, cmd->line, len);
+            str[len] = '\0';
+        } else {
+            str[len] = cmd->line[0];
+        }
```

Comment:
This is broken. The common case, where cnt > len, would copy no bytes to str.

Also, I don't understand line 96 here. Why are you setting last character in str to be
first character of input?